src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntry.LongFile.Tests.cs

```diff
@@ -86,6 +86,7 @@ public void WriteEntry_LongFileSize(TarEntryFormat entryFormat, long size, bool
             Assert.Equal(size, dataStream.Position);
 
             Assert.Null(reader.GetNextEntry());
+            tarFile.Close();
```

Comment:
I would expect #79899 to solve this problem for good. If that is not the case, we need to investigate it and ensure that `Dispose` does the right thing.

Reply:
it didn't solve the issue for me, at least on Mono. I haven't checked on CoreCLR (GC behavior can be quite different between the runtimes)

Reply:
hmm, interesting: without the change in this PR it looks like the file does get deleted on disk however the process still keeps it open at least according to Activity Monitor on Mac:

<img width="1372" alt="image" src="https://user-images.githubusercontent.com/1376924/209142833-6a041e74-199c-4785-89e7-fb5c1bb4fe80.png">

and it also still counts against the `Available` disk space in `df -u`.

Reply:
seems to be specific to the Async test

Reply:
ahhh the fix in https://github.com/dotnet/runtime/pull/79899 didn't get applied to the `DisposeAsync()` method 😆

Reply:
ok and another problem is that the `WrappedStream` isn't disposing the inner `_baseStream`. I'll be away for a few hours now, but I can send a PR later.

Reply:
@akoeplinger thank you for investigating it! 

> I can send a PR later.

That would be perfect.

Reply:
see https://github.com/dotnet/runtime/pull/79920